public/src/managers/ServerProvider.ts

```diff
@@ -84,9 +84,6 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
     public async getCourses(): Promise<Course[]> {
         const result = await this.grpcHelper.getCourses();
         if (result.statusCode !== 0 || !result.data) {
-            if (result.message) {
```

Comment:
Nice update to avoid these. I notice that there are some variations in that some methods check only the status code and others also check if there is `result.data`. It is not clear to me if these variations are required or if it is a bug. If it is a bug, I suggest to add a check function `hasResult` that returns a boolean that can be used everywhere. Maybe the result objects themselves can have such a function. That is, only if the same check function can be used for all result types, as otherwise we would just move the problem elsewhere.

On the other hand, I wonder if the return of empty list early is actually needed, since the iteration over the result list would be empty and so unless `result` is `null`, the `map` and other code may just return the empty list also, making the check unnecessary. (This would be a nice instance of defining errors out of existence ;-)

Reply:
Not a bug. Some grpc methods are only supposed to return errors/status codes. In such methods `result.data` will be `null` or of type Void (kinda `null` but sugarcoated for protobuf needs) and we only care about status code (thats where error notification for user is coming from).